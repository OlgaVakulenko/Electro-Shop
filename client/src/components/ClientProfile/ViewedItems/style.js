import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles((theme) => ({
  expansionPanelDetails: {
    [theme.breakpoints.down(600)]: {
      background: 'linear-gradient(180deg, #F8F8F8 0%, rgba(248, 248, 248, 0) 100%)',
    },
  },
  heading: {
    fontWeight: '600',
    flexBasis: '66.66%',
    flexShrink: 0,
  },
  mainContainer: {
    paddingTop: 30,
    paddingBottom: 30,
  },
  card: {
    marginBottom: 0,
    maxWidth: 'auto',
  },
  sectionBackground: {
    backgroundColor: '#FAFAFA',
  },
}));
