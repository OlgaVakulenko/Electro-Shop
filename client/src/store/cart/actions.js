import axios from 'axios';

// ACTIONS WITH CART

export const ADD_PRODUCT_TO_CART = 'ADD_PRODUCT_TO_CART';

export const addProductToCart = (url) => (dispatch) => {
	axios.put(url)
		.then(cart => {
			dispatch({
				type: ADD_PRODUCT_TO_CART,
				payload: cart.data.products,
			})
		})
};