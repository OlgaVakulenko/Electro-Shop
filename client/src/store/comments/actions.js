import axios from 'axios';
// import { enqueueSnackbar } from '../notification/actions';

// ACTIONS WITH COMMENTS
export const ADD_COMMENT = 'ADD_COMMENT';
export const GET_COMMENTS = 'GET_COMMETS';

// ACTIONS
export const addComment = (comment, id) => (dispatch) => {
  axios.post('/comments', comment)
    .then(() => {
      axios
        .get(`/comments/product/${id}`)
        .then((data) => {
          if (data.data) {
            dispatch({
              type: GET_COMMENTS,
              payload: data.data,
            });
          }
        });
    });
};

export const getComments = (id) => (dispatch) => {
  axios
    .get(`/comments/product/${id}`)
    .then((data) => {
      if (data.data) {
        dispatch({
          type: GET_COMMENTS,
          payload: data.data,
        });
      }
    });
};
