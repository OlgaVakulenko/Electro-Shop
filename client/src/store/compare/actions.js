export const ADD_PRODUCTS_TO_COMPARE = 'ADD_PRODUCTS_TO_COMPARE';
export const DELETE_PRODUCTS_FROM_COMPARE = 'DELETE_PRODUCTS_FROM_COMPARE';

