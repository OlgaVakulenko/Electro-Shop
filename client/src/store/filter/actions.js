import axios from 'axios';
import { FILTER_ACTIONS } from './reducers';

// action:
export const getFilters = (payload) => ({
  type: FILTER_ACTIONS.GET_FILTERS,
  payload,
});
export const getFilers = (payload) => ({
  type: FILTER_ACTIONS.SET_FILTERS,
  payload,
});

function reducer(store = initialStore, { type, payload }) {
  switch (type) {
    case GET_FILTERS:
      return {
        ...store,
        filters: payload,

      };
    case SET_FILTERS:
      return {
        ...store,
        selectedFilters: payload,
      };
    default:
      return store;
  }
}
const store = createStore(
  reducer, applyMiddleware(thunk),
);
// store.dispatch(setAuth(true));

export default store;
