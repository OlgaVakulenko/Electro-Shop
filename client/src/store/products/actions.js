import axios from 'axios';

// ACTIONS WITH OTHER FILTERS
export const GET_PRODUCTS_REQUESTED = 'GET_PRODUCTS_REQUESTED';
export const GET_PRODUCTS_SUCCEEDED = 'GET_PRODUCTS_SUCCEEDED';
export const GET_PRODUCTS_FAILED = 'GET_PRODUCTS_FAILED';
export const GET_MORE_PRODUCTS_REQUESTED = 'GET_MORE_PRODUCTS_REQUESTED';
export const GET_MORE_PRODUCTS_SUCCEEDED = 'GET_MORE_PRODUCTS_SUCCEEDED';
export const GET_MORE_PRODUCTS_FAILED = 'GET_MORE_PRODUCTS_FAILED';
export const CLEAR_NEW_PRODUCTS = 'CLEAR_NEW_PRODUCTS';

// action:
export const getProducts = (endpoint) => (dispatch) => {
  console.log('ACTION запущен экшен getProducts');
  dispatch({
    type: GET_PRODUCTS_REQUESTED,
  });
  axios.get(endpoint)
    .then((data) => {
      console.log('ACTION запущен экшен getProducts - data в ответет сервера', data);
      const allProducts = data.data.products.map((el) => ({
        itemNo: el.itemNo,
        name: el.name,
        itemImg: el.imageUrls[0],
        price: el.currentPrice,
        url: `/products/${el.itemNo}`,
        rating: el.rating,
        id: el._id,
        distance: el.distance,
        maxSpeed: el.maxSpeed,
        chargingTime: el.chargingTime,
      }));
      console.log('ACTION allProducts', allProducts);
      dispatch({
        type: GET_PRODUCTS_SUCCEEDED,
        allProducts,
      });
    })
    .catch((err) => {
      dispatch({
        type: GET_PRODUCTS_FAILED,
        payload: err,
      });
    });
};

export const getMoreProducts = (endpoint) => (dispatch) => {
  dispatch({
    type: GET_MORE_PRODUCTS_REQUESTED,
  });
  axios.get(endpoint)
    .then((data) => {
      const newProducts = data.data.products.map((el) => ({
        itemNo: el.itemNo,
        name: el.name,
        itemImg: el.imageUrls[0],
        price: el.currentPrice,
        url: `/products/${el.itemNo}`,
        rating: el.rating,
        id: el._id,
        distance: el.distance,
        maxSpeed: el.maxSpeed,
        chargingTime: el.chargingTime,
      }));
      dispatch({
        type: GET_MORE_PRODUCTS_SUCCEEDED,
        newProducts,
      });
    })
    .catch((err) => {
      dispatch({
        type: GET_MORE_PRODUCTS_FAILED,
        payload: err,
      });
    });
};
export const clearNewProducts = () => (dispatch) => {
  dispatch({
    type: CLEAR_NEW_PRODUCTS,
    newProducts: [],
  });
}