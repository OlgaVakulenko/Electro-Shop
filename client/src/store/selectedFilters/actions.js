export const SELECT_FILTERS = 'SELECT_FILTERS';

export const selectFilters = (event, value, type) => (dispatch) => {
  dispatch({
    type: SELECT_FILTERS,
    selectedFilters: [{
      selectedFilter: value,
      filterType: type,
    }],
  });
};
