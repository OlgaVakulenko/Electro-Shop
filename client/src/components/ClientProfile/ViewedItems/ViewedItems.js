/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
// Material UI
import {
  Container, ExpansionPanel, ExpansionPanelDetails, ExpansionPanelSummary, Typography,
} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
// Local imports
import { useStyles } from './style';
import { addProductAndCreateWishlistInDB } from '../../../store/wishlist/actions';
import ProductCard from '../../ProductCard/ProductCard';

const ViewedItems = (props) => {
  const classes = useStyles();
  const [expanded, setExpanded] = useState('');
  const [productsList, setProductsList] = useState([]);
  const [arrInfinity, setArrInfinity] = useState({ infinity: false });
  const { addProductToWishlist } = props;
  const handleChange = (panel) => (event, isExpanded) => {
    setExpanded(isExpanded ? panel : false);
  };

  let products;
  let productsListInfo;
  const currentLocal = JSON.parse(localStorage.getItem('product'));

  useEffect(() => {
    setProductsList(currentLocal);
    if (currentLocal && currentLocal.length > 1) {
      setArrInfinity({ infinity: true });
    }
  }, []);

  const settings = {
    centerPadding: '0px',
    dots: true,
    infinite: arrInfinity.infinity,
    speed: 650,
    slidesToShow: 2,
    slidesToScroll: 1,
    initialSlide: 0,
    autoplay: true,
    pauseOnHover: true,
    arrows: false,
    responsive: [
      {
        breakpoint: 1052,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 482,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          centerMode: true,
        },
      },
    ],
  };

  if (productsList) {
    products = productsList.map((el) => (
      <div
        key={el.itemNo}
      >
        <ProductCard
          obj={el.obj}
          className={classes.card}
          name={el.name}
          itemImg={el.itemImg}
          price={el.price}
          url={el.url}
          rating={el.rating}
          itemNo={el.itemNo}
          id={el.id}
          addProductToWishlist={addProductToWishlist}
        />
      </div>
    ));
  }

  if (productsList) {
    productsListInfo = (
      <Slider
        {...settings}
        className={classes.paddingTop}
      >
        {products}
      </Slider>
    );
  } else {
    productsListInfo = (
      <Typography
        align="center"
      >
        You have not seen any item yet.
      </Typography>
    );
  }

  return (
    <React.Fragment>
      <ExpansionPanel
        expanded={expanded === 'panel4'}
        onChange={handleChange('panel4')}
      >
        <ExpansionPanelSummary
          expandIcon={<ExpandMoreIcon />}
          aria-controls="panel4bh-content"
          id="panel4bh-header"
        >
          <Typography
            className={classes.heading}
            style={{ color: '#6A86E8' }}
          >
            My Viewed Items
          </Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails
          className={classes.expansionPanelDetails}
        >
          <Container
            className={classes.mainContainer}
          >
            {productsListInfo}
          </Container>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    </React.Fragment>
  );
};

const ViewedItms = connect(null, {
  addProductToWishlist: addProductAndCreateWishlistInDB,
})(ViewedItems);
export { ViewedItms as ViewedItems };
