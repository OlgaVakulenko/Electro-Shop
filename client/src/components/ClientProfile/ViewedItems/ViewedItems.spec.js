import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { ViewedItems } from './ViewedItems';
import { addProductAndCreateWishlistInDB } from '../../../store/wishlist/actions';
const mockStore = configureStore([]);

describe('The test of ViewedItems component', () => {
  let store;

  beforeAll(() => {
    store = mockStore(null,
      {
        addProductToWishlist: addProductAndCreateWishlistInDB,
      });
  });

  it('ViewedItems was rendered', () => {
    const rendered = shallow(
      <Provider store={store}>
        <ViewedItems />
      </Provider>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
