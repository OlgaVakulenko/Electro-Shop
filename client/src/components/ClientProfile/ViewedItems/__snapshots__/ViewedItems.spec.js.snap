// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test of ViewedItems component ViewedItems was rendered 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
  >
    <WithStyles(ForwardRef(ExpansionPanelSummary))
      aria-controls="panel4bh-content"
      expandIcon={<Memo(ExpandMoreIcon) />}
      id="panel4bh-header"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-heading-2"
        style={
          Object {
            "color": "#6A86E8",
          }
        }
      >
        My Viewed Items
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ExpansionPanelSummary))>
    <WithStyles(ForwardRef(ExpansionPanelDetails))
      className="makeStyles-expansionPanelDetails-1"
    >
      <WithStyles(ForwardRef(Typography))>
        Nunc vitae orci ultricies, auctor nunc in, volutpat nisl. Integer sit amet egestas eros, vitae egestas augue. Duis vel est augue.
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(ExpansionPanelDetails))>
  </WithStyles(ForwardRef(ExpansionPanel))>
</Fragment>
`;
